src/ig/IGExporter.ts

```diff
@@ -405,6 +407,77 @@ export class IGExporter {
     });
   }
 
+  /**
+   * Adds any user provided resource files
+   * This includes definitions in:
+   * capabilities, extensions, models, operations, profiles, resources, vocabulary, examples
+   *
+   * @param {string} igPath - the path where the IG is exported to
+   */
+  private addPredefinedResources(igPath: string): void {
+    const pathEnds = [
+      'capabilities',
+      'extensions',
+      'models',
+      'operations',
+      'profiles',
+      'resources',
+      'vocabulary',
+      'examples' // Must come last in case examples are of other resources
+    ];
+    for (const pathEnd of pathEnds) {
+      const dirPath = path.join(this.igDataPath, 'input', pathEnd);
+      if (fs.existsSync(dirPath)) {
+        const files = fs.readdirSync(dirPath);
+        for (const file of files) {
+          let resourceJSON: InstanceDefinition;
+          try {
+            resourceJSON = JSON.parse(fs.readFileSync(path.join(dirPath, file), 'utf-8').trim());
+          } catch (e) {
+            logger.error(`Invalid input file: ${path.join(dirPath, file)}`);
+            continue;
+          }
+          if (resourceJSON.resourceType == null || resourceJSON.id == null) {
+            logger.error(`Resource at ${path.join(dirPath, file)} must define resourceType and id`);
+            continue;
+          }
+          const resource: ImplementationGuideDefinitionResource = {
+            reference: {
+              reference: `${resourceJSON.resourceType}/${resourceJSON.id}`
+            },
+            name: resourceJSON.title ?? resourceJSON.id,
+            description: resourceJSON.description
+          };
+          if (pathEnd === 'examples') {
+            const exampleUrl = resourceJSON.meta?.profile?.find(
+              url =>
+                this.fhirDefs.fishForFHIR(url, Type.Profile) ?? this.pkg.fish(url, Type.Profile)
+            );
+            if (exampleUrl) {
```

Comment:
This check isn't strict enough because we should only set `exampleCanonical` for examples of profiles _defined in this IG_.  Since you're searching through all of FHIRDefs, it's possible that it matched a profile from a dependency IG -- and in that case we should only use `exampleBoolean`, _not_ `exampleCanonical`.